### PR TITLE
Fix GHE Error Handling

### PR DESCRIPTION
**Pull request by mitchute**
Pull request overview
---------------------
 - Fixes GHE error handling for getter routines

### CODE REVIEW

File: src/EnergyPlus/GroundHeatExchangers.cc [hunk 1]
```diff
@@ -320,11 +320,6 @@ GLHEVert::GLHEVert(EnergyPlusData &state, std::string const &objName, nlohmann::
         this->myRespFactors =
             GetResponseFactor(state, UtilityRoutines::MakeUPPERCase(j["ghe_vertical_responsefactors_object_name"].get<std::string>()));
         this->gFunctionsExist = true;
-
-        if (!this->myRespFactors) {
-            errorsFound = true;
-            ShowSevereError(state, "GroundHeatExchanger:ResponseFactors object not found.");
-        }
```
Comment (mitchute) [inline src/EnergyPlus/GroundHeatExchangers.cc:327]: Unnecessary error check at the call site.

File: src/EnergyPlus/GroundHeatExchangers.cc [hunk 2]
```diff
@@ -348,11 +343,6 @@ GLHEVert::GLHEVert(EnergyPlusData &state, std::string const &objName, nlohmann::
             // Response factors come from array object
             this->myRespFactors = BuildAndGetResponseFactorObjectFromArray(
                 state, GetVertArray(state, UtilityRoutines::MakeUPPERCase(j["ghe_vertical_array_object_name"].get<std::string>())));
-
-            if (!this->myRespFactors) {
-                errorsFound = true;
-                ShowSevereError(state, "GroundHeatExchanger:Vertical:Array object not found.");
-            }
```
Comment (mitchute) [inline src/EnergyPlus/GroundHeatExchangers.cc:355]: Same

File: src/EnergyPlus/GroundHeatExchangers.cc [hunk 3]
```diff
@@ -369,14 +359,7 @@ GLHEVert::GLHEVert(EnergyPlusData &state, std::string const &objName, nlohmann::
...
                     std::shared_ptr<GLHEVertSingle> tempBHptr =
                         GetSingleBH(state, UtilityRoutines::MakeUPPERCase(var.at("ghe_vertical_single_object_name").get<std::string>()));
-                    if (tempBHptr) {
-                        tempVectOfBHObjects.push_back(tempBHptr);
-                    } else {
-                        errorsFound = true;
-                        std::string const tmpName = var.at("ghe_vertical_single_object_name").get<std::string>();
-                        ShowSevereError(state, "Borehole= " + tmpName + " not found.");
-                        break;
-                    }
```
Comment (mitchute) [inline src/EnergyPlus/GroundHeatExchangers.cc:379]: Same

File: src/EnergyPlus/GroundHeatExchangers.cc [hunk 4]
```diff
@@ -589,6 +572,10 @@ std::shared_ptr<GLHEVertProps> GetVertProps(EnergyPlusData &state, std::string c
         }
     }
 
+    ShowSevereError(state, fmt::format("Object=GroundHeatExchanger:Vertical:Properties, Name={} - not found.", objectName));
+    ShowFatalError(state, "Preceding errors cause program termination");
```
Comment (mitchute) [inline src/EnergyPlus/GroundHeatExchangers.cc:576]: Error handling in the getter function, where it should be.

File: src/EnergyPlus/GroundHeatExchangers.cc [hunk 5]
```diff
@@ -604,6 +591,10 @@ std::shared_ptr<GLHEVertSingle> GetSingleBH(EnergyPlusData &state, std::string c
         }
     }
 
+    ShowSevereError(state, fmt::format("Object=GroundHeatExchanger:Vertical:Single, Name={} - not found.", objectName));
+    ShowFatalError(state, "Preceding errors cause program termination");
```
Comment (mitchute) [inline src/EnergyPlus/GroundHeatExchangers.cc:595]: Same

File: tst/EnergyPlus/unit/GroundHeatExchangers.unit.cc [hunk 6]
```diff
@@ -4236,3 +4236,313 @@ TEST_F(EnergyPlusFixture, GroundHeatExchangerTest_System_calcBHTotalInternalResi
     EXPECT_NEAR(thisGLHE.sigma, (thisGLHE.grout.k - thisGLHE.soil.k) / (thisGLHE.grout.k + thisGLHE.soil.k), tolerance);
     EXPECT_NEAR(thisGLHE.calcBHTotalInternalResistance(*state), 0.31582, tolerance);
 }
+
+TEST_F(EnergyPlusFixture, GroundHeatExchangerTest_GetVertProps)
+{
+    std::string const idf_objects = delimited_string({
+        "GroundHeatExchanger:Vertical:Properties,",
+        "    GHE-1 Props,        !- Name",
```
Comment (mitchute) [inline tst/EnergyPlus/unit/GroundHeatExchangers.unit.cc:4244]: New unit tests

**Review by mitchute — COMMENTED** [review]
GetInput error handling relied on the proper treatment of errors at all getter-method call sites. However, I realized that all of the call sites were not taking care of potential errors, so I refactored this so the getter methods take care of their own errors.

**Comment by Myoldmopar** [comment]
Better error handling, changes contained in the GHE code, new unit tests, super clean CI.  This seems ready to go.  Any objections anyone?

**Comment by Myoldmopar** [comment]
Also this branch is already up to date with develop, so I'll merge it shortly if no objections.

**Pull request merged**